src/components/PostDetails.tsx

```diff
@@ -1,106 +1,121 @@
-import React from 'react';
-import { Loader } from './Loader';
+import React, { useEffect } from 'react';
 import { NewCommentForm } from './NewCommentForm';
+import { Loader } from './Loader';
+import { Comment, CommentData } from '../types/Comment';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+type Props = {
+  isCommentLoading: boolean;
+  postComments: Comment[];
+  selectedPost: Post | null;
+  errorsComment: boolean;
+  onDeleteComment: (commentId: number) => void;
+  onAddComment: (newComment: CommentData) => void;
+  isButtonAddLoading: boolean;
+  onShowCommentFormChange: (item: boolean) => void;
+  isCommentFormVisible: boolean;
+  userSelected: User | null;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  isCommentLoading,
+  postComments,
+  selectedPost,
+  errorsComment,
+  onDeleteComment,
+  onAddComment,
+  isButtonAddLoading,
+  onShowCommentFormChange,
+  isCommentFormVisible,
+  userSelected,
+}) => {
+  useEffect(() => {
+    onShowCommentFormChange(false);
```

Comment:
You are initializing your state with props. This can lead to bugs and inconsistencies in your application state. It's better to use props directly or use a state management library like Redux or MobX. Check these articles for more details: https://stackoverflow.com/a/50403930, https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state, https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state